core/src/main/scala/cats/data/NonEmptyChain.scala

```diff
@@ -22,14 +22,12 @@
 package cats
 package data
 
-import NonEmptyChainImpl.create
-
 import cats.instances.StaticMethods
 import cats.kernel._
 
 import scala.collection.immutable.SortedMap
 
-private[data] object NonEmptyChainImpl extends NonEmptyChainInstances with ScalaVersionSpecificNonEmptyChainImpl {
+private[data] object NonEmptyChainImpl extends NonEmptyChainImpl {
```

Comment:
Btw I suppose we could also just make this object public, instead of extracting its methods into a public class.

Reply:
Yeah, I thought about it too. But in this case anyone can import and access the `NonEmptyChainImpl` object directly, which should not be allowed, I think.

Reply:
Fair point. I guess someone could still import `sealed abstract class NonEmptyChainImpl`, but it's harder to do anything with it.

Reply:
In light of https://github.com/typelevel/cats/pull/4219#discussion_r889607029, I think we should just make these objects public. Seems like it will get us bincompat checking through Cats 1.x which I'd say is a win.

Reply:
But should we really worry about bin-compat with 1.x version?

Reply:
Let me clarify: bincompat checking through 1.x means all the way from 1.0.0-2.7.0. So even if you don't care about 1.x, it gets us 2.x, which is still a win in my book :)

Reply:
@satorg any new thoughts about this? Would be nice to land this before 2.8.0 but it's not a blocker.

Reply:
Sorry, could not pay enough attention to it this week... But I'm still a bit hesitant of making all these `NonEmpty*Impl` object just public.. I wouldn't be so concerned if these object were located somewhere inside the `internal` package. But if we make them public inside `cats.data` – won't we worsen the _source compatibility_ instead? I mean, it would become possible (and legit) to reference these `*Impl` objects directly bypassing the suggested `val` references.

Reply:
No worries. I completely agree, but I don't think it is that much worse than the changes you are proposing here.

Basically, this was a newtype hack, and there is a trade-off here between cleanliness of the source vs how easy it is to maintain bincompat. Indeed, we've already broken bincompat here without realizing it.

Reply:
ok, I'll change them to just public objects shortly.

Just curious – how come these `NonEmpty*Impl` were introduced at all? I mean, to me it seems like initially it should be possible just to create `NonEmpty*` objects in place of their `val` references. Now it is not the case of course due the binary compatibility, but at the time they were created they could be created in that way (I guess).

Reply:
@armanbilge done, take a look when u have time please.